models/sipnet/R/model2netcdf.SIPNET.R

```diff
@@ -48,8 +55,28 @@ model2netcdf.SIPNET <- function(outdir, sitelat, sitelon, start_date, end_date,
     sub.sipnet.output <- subset(sipnet.output, year == y)
     sub.sipnet.output.dims <- dim(sub.sipnet.output)
     dayfrac <- 1 / out.day
-    step <- seq(0, 0.99, dayfrac)
-
+    step <- seq(0, 0.99, dayfrac)  ## probably dont want to use hard-coded "step" because leap years may not contain all "steps", or
+                                   ## if model run doesnt start at 00:00:00
+    
+    # get the run dates based on the sipnet output.  we assume that even if the run is partial, the origin is still day 1 of the subset year
+    start_month <- lubridate::month(as.Date(sub.sipnet.output$day[1], origin = paste0(y,"-01-01"))) # gets month
+    ## using this approach to attempt to deal with inconsistent start index 0/1
+    model_start_date <- base::as.Date(paste0(y,"-",start_month,"-",ifelse(sub.sipnet.output$day[1]==0,1,sub.sipnet.output$day[1])))  
+    sub_date_range <- seq(model_start_date, by = "day", length.out = length(unique(sub.sipnet.output$day))) ## create new date range in POSIX format
+    # this catches the fact that the number of outputs per day may be different at the end of the year with leap years, if day starts at 1
+    day_repeats <- as.vector(base::table(sub.sipnet.output$day))  
```

Comment:
@ashiklom I would argue that met process is failing *despite* being far more generalized and generic than the generally one-off model2netCDF.X() functions that share some features/code tweaked to fir the model-specific outputs.  So I agree but I would not say they are equivalent and in fact with MP it looks more like disconnect between how files/inputs are registered in the DB and then re-used within the functions.

Reply:
Fair enough. Per our discussion on Slack, I'm fine leaving this as is, but let's open an issue to keep it in mind. Importantly, if there are any key differences (i.e. SIPNET-specific things) between what's going on here and in the same code for ED, then it's worth mentioning those in a comment (and/or in the issue), just anyone considering refactoring this in the future doesn't miss them.

Reply:
@ashiklom Agreed....BUT I would also say that a person creating an issue such as this may not know all of the key differences between the two functions, and if so, then whom does that burden fall?  The originator (that may be hard to track down now) or...???